lib/Doctrine/DBAL/Migrations/Tools/Console/Command/DiffCommand.php

```diff
@@ -109,7 +109,7 @@ public function execute(InputInterface $input, OutputInterface $output)
         }
 
         $version = date('YmdHis');
-        $path = $this->generateMigration($configuration, $input, $version, $up, $down);
+        $path = $this->generateMigration($configuration, $input, $version, $up, $down, true);
```

Comment:
Totally separate issue from this method, but again it's doing exactly one thing (executing a migration). It would be nicer, IMO, to just give the migration itself a method that turned the migration into executable SQL statements. But that's a pretty major refactor that doesn't make a lot of sense with the current system and would cause a lot of BC breaks.

Idealism is great, but not practical in this case. It's not an idea solution, but it is a solution that would provide a benefit for a lot of people.